sql/driver/driver_test.go

```diff
@@ -45,18 +49,611 @@ func cleanup(s *server.TestServer, db *sql.DB) {
+		"Europe/Kiev",
+		"Europe/Lisbon",
+		"Europe/Ljubljana",
+		"Europe/London",
+		"Europe/Luxembourg",
+		"Europe/Madrid",
+		"Europe/Malta",
+		"Europe/Mariehamn",
+		"Europe/Minsk",
+		"Europe/Monaco",
+		"Europe/Moscow",
+		"Europe/Nicosia",
+		"Europe/Oslo",
+		"Europe/Paris",
+		"Europe/Podgorica",
+		"Europe/Prague",
+		"Europe/Riga",
+		"Europe/Rome",
+		"Europe/Samara",
+		"Europe/San_Marino",
+		"Europe/Sarajevo",
+		"Europe/Simferopol",
+		"Europe/Skopje",
+		"Europe/Sofia",
+		"Europe/Stockholm",
+		"Europe/Tallinn",
+		"Europe/Tirane",
+		"Europe/Tiraspol",
+		"Europe/Uzhgorod",
+		"Europe/Vaduz",
+		"Europe/Vatican",
+		"Europe/Vienna",
+		"Europe/Vilnius",
+		"Europe/Volgograd",
+		"Europe/Warsaw",
+		"Europe/Zagreb",
+		"Europe/Zaporozhye",
+		"Europe/Zurich",
+		"GB",
+		"GB-Eire",
+		"GMT",
+		"GMT+0",
+		"GMT0",
+		"GMT-0",
+		"Greenwich",
+		"Hongkong",
+		"Iceland",
+		"Indian/Antananarivo",
+		"Indian/Chagos",
+		"Indian/Christmas",
+		"Indian/Cocos",
+		"Indian/Comoro",
+		"Indian/Kerguelen",
+		"Indian/Mahe",
+		"Indian/Maldives",
+		"Indian/Mauritius",
+		"Indian/Mayotte",
+		"Indian/Reunion",
+		"Iran",
+		"Israel",
+		"Jamaica",
+		"Japan",
+		"Kwajalein",
+		"Libya",
+		"Mexico/BajaNorte",
+		"Mexico/BajaSur",
+		"Mexico/General",
+		"Navajo",
+		"NZ",
+		"NZ-CHAT",
+		"Pacific/Apia",
+		"Pacific/Auckland",
+		"Pacific/Chatham",
+		"Pacific/Chuuk",
+		"Pacific/Easter",
+		"Pacific/Efate",
+		"Pacific/Enderbury",
+		"Pacific/Fakaofo",
+		"Pacific/Fiji",
+		"Pacific/Funafuti",
+		"Pacific/Galapagos",
+		"Pacific/Gambier",
+		"Pacific/Guadalcanal",
+		"Pacific/Guam",
+		"Pacific/Honolulu",
+		"Pacific/Johnston",
+		"Pacific/Kiritimati",
+		"Pacific/Kosrae",
+		"Pacific/Kwajalein",
+		"Pacific/Majuro",
+		"Pacific/Marquesas",
+		"Pacific/Midway",
+		"Pacific/Nauru",
+		"Pacific/Niue",
+		"Pacific/Norfolk",
+		"Pacific/Noumea",
+		"Pacific/Pago_Pago",
+		"Pacific/Palau",
+		"Pacific/Pitcairn",
+		"Pacific/Pohnpei",
+		"Pacific/Ponape",
+		"Pacific/Port_Moresby",
+		"Pacific/Rarotonga",
+		"Pacific/Saipan",
+		"Pacific/Samoa",
+		"Pacific/Tahiti",
+		"Pacific/Tarawa",
+		"Pacific/Tongatapu",
+		"Pacific/Truk",
+		"Pacific/Wake",
+		"Pacific/Wallis",
+		"Pacific/Yap",
+		"Poland",
+		"Portugal",
+		"PRC",
+		"ROC",
+		"ROK",
+		"Singapore",
+		"Turkey",
+		"UCT",
+		"Universal",
+		"US/Alaska",
+		"US/Aleutian",
+		"US/Arizona",
+		"US/Central",
+		"US/Eastern",
+		"US/East-Indiana",
+		"US/Hawaii",
+		"US/Indiana-Starke",
+		"US/Michigan",
+		"US/Mountain",
+		"US/Pacific",
+		"US/Samoa",
+		"UTC",
+		"W-SU",
+		"Zulu",
+	}
+
 	s, db := setup(t)
 	defer cleanup(s, db)
 
-	year, month, day := 3015, time.August, 30
-	timeVal := time.Date(year, month, day, 3, 34, 45, 345670000, time.UTC)
-	dateVal := time.Date(year, month, day, 0, 0, 0, 0, time.UTC)
+	offsets := make(map[int]struct{})
+
+	// We don't want to spend forever on this test.
+	for startTime := time.Now(); time.Since(startTime) < 10*time.Second; {
```

Comment:
Ten seconds is a long time for one test (and this even devolves into a busy loop if we've tried every time zone so it will always take the full ten seconds). Instead of random sampling from a nearly-complete list of time zones ("nearly-complete" because new zones are added every year; the most recent addition being America/Fort_Nelson), we should just pick a few interesting zones and run with those. For example:
* UTC
* Europe/London (which is the same offset as UTC on standard time and differs in DST)
* One zone with a negative offset and one with a positive offset
* One of the half-hour zones
* Pacific/Kiritimati (the highest absolute offset at +14:00)